plugins/httpinspector.koplugin/main.lua

```diff
@@ -0,0 +1,1267 @@
+        end
+        if value_type == "userdata" then
+            return "userdata"
+        end
+    end
+    -- May fail with recursive data
+    local JSON = require("json")
+    return JSON.encode(obj, encoder_options)
+end
+
+-- We cache the results of the next function, to avoid opening and reading
+-- each source file each time
+local _function_info_cache = {}
+
+-- Get info about a function object
+local getFunctionInfo = function(func, full_code)
+    local info = debug.getinfo( func, "S" )
+    local src, firstline, lastline = info.source, info.linedefined, info.lastlinedefined
+    local hash = src.."#"..firstline.."#"..lastline
+    if _function_info_cache[hash] and not full_code then
+        return _function_info_cache[hash]
+    end
+    local path = src:match("^@(.*)$")
+    local lines = nil
+    if path and info.what == "Lua" then
+        local f = io.open(path)
+        if f then
+            local num = 1
+            while true do
+                local line = f:read('*line')
+                if not line then
+                    break
+                end
+                if num >= firstline then
+                    if not lines then
+                        lines = {}
+                    end
+                    table.insert(lines, line)
+                end
+                if num >= lastline then
+                    break
+                end
+                num = num + 1
+            end
+            f:close()
+        end
+    end
+    info = {
+        source = path,
+        firstline = firstline,
+        lastline = lastline,
+    }
+    if lines then
+        local signature = util.trim(lines[1])
+        info.signature = signature
+        -- Try to guess (possibly wrongly) a few info from the signature string
+        local dummy, cnt
+        dummy, cnt = signature:gsub("%(%)","") -- check for "()", no arg
+        if cnt > 0 then
+            info.nb_args = 0
+        else
+            dummy, cnt = signature:gsub(",","") -- check for nb of commas
+            info.nb_args = cnt and cnt + 1 or 1
+        end
+        dummy, cnt = signature:gsub("%.%.%.","") -- check for "...", varargs
+        if cnt > 0 then
+            info.nb_args = -1
+        end
+        dummy, cnt = signature:gsub("^[^(]*:","")
+        info.is_method = cnt > 0
+        info.classname = signature:gsub(".-(%w+):.*","%1")
+    end
+    _function_info_cache[hash] = info
+    if not full_code then
+        return info
+    end
+    info = util.tableDeepCopy(info)
+    info.lines = lines
+    return info
+end
+
+-- Guess class name of an object
+local guessClassName = function(obj)
+    -- Look for some common methods we could infer a class from (add more as needed)
+    local classic_method_names = {
+        "init",
+        "new",
+        "getSize",
+        "paintTo",
+        "onReadSettings",
+        "onMenuHold",
+        "beforeSuspend",
+        "initNetworkManager",
+        "free",
+        "clear",
```

Comment:
Adding "onResume" allows detecting DeviceListener and NetworkListener.

Reply:
But gives garbage in BackgroundRunner entry in FileManager ui.

Reply:
We get such garbage elsewhere for all plugins onXYZ methods, as they are wrapped by pluginloader :/
https://github.com/koreader/koreader/blob/bdd475f55f9fe0530cbc3fd9825814831bdd86f7/frontend/pluginloader.lua#L28-L32

(I tried to keep the original method with another name, but it got messy with multiple plugin instantiations :) anyway, I ended up thinking this plugin should may be not do to many adjustments, and just keep telling the truth - even if the truth is ugly sometimes :))

Reply:
There are also other modules that have only unique methods, not shared with anything else, ie. ReaderUserHyph.
If we're annoyed with not seeing their name, we could just plug some dummy methods in the classes that need it, ie `function ReaderUserHyph:_forClassIdentification() return`